bson-adt-core/src/main/scala/adt/bson/BsonAdtImplicits.scala

```diff
@@ -0,0 +1,85 @@
+package adt.bson
+
+import adt.bson.mongo.codecs.{BsonAdtCodecRegistry, BsonAdtCodec}
+import org.bson.codecs.configuration.CodecRegistry
+import org.bson.codecs.{DecoderContext, EncoderContext}
+import org.bson.conversions.{Bson => BsonConversion}
+import org.bson.{BsonDocumentReader, BsonDocumentWriter}
+
+import scala.collection.JavaConversions.seqAsJavaList
+import scala.language.{higherKinds, implicitConversions}
+
+/**
+ * Adds implicit conversions between [[JavaBsonValue]] and [[BsonValue]].
+ *
+ * @note these are provided when you `import adt.bson.mongo._`
+ */
+trait BsonAdtImplicits {
+
+  implicit def bsonValueAsJavaOps(adt: BsonValue): BsonAdtValueOps = new BsonAdtValueOps(adt)
```

Comment:
Implicit class is just a syntax sugar for this. Unfortunately, implicit classes don't work as seemlessly with value classes as they probably should. The rules are:

1. Value classes cannot live inside classes, but they can living inside singleton objects
2. Implicit classes must live inside another class or object.

Since I wanted the ability to inherit these implicits, and I wanted value classes, I can't benefit from the syntax sugar of implicit classes.